src/components/user/PersonalProfileEditForm.js

```diff
@@ -0,0 +1,205 @@
+ 
+  const handleInputChange = (e) => {
+    const { name, value } = e.target;
+    setFormData({ ...formData, [name]: value });
+  };
+
+  const handleProfileSubmit = async (e) => {
+    e.preventDefault();
+    if (!formData.name || !formData.email) {
+      // Check if either name or email is empty
+      setErrors({
+        name: !formData.name ? "Name is required" : "",
+        email: !formData.email ? "Email is required" : "",
+      });
+    } else {
+      try {
+        // Call the backend API to update the user's name and email
+        const response = await axiosInstance.patch('/users/updateCurrentUser', {
+          name: formData.name,
+          email: formData.email,
+        });
+        // Handle success (e.g., show a success message)
+        console.log('User profile updated:', response.data);
+        setErrors({}); // Clear any previous errors
+        setSuccessMessage('User profile saved successfully.'); // Set success message
+        setFormData({ // Clear the form
+          name: '',
+          email: '',
+          oldPassword: '',
+          newPassword: '',
+        });
+        // Navigate back to the user profile page
+        navigate('/profile'); // Use the navigate function
+      } catch (error) {
+        // Handle errors and update the state with validation errors
+        setErrors(error.response.data.msg);
+        <Alert severity="error">{errors}</Alert>
+        setSuccessMessage(''); // Clear any previous success message
+      }
+    }
+  };
+
+  const handlePasswordSubmit = async (e) => {
+    e.preventDefault();
+    if (!formData.oldPassword || !formData.newPassword) {
+      // Check if either oldPassword or newPassword is empty
+      setErrors({
+        oldPassword: !formData.oldPassword ? "Old Password is required" : "",
+        newPassword: !formData.newPassword ? "New Password is required" : "",
+      });
+    } else {
+      try {
+        // Call the backend API to update the user's password
+        const response = await axiosInstance.patch('/users/updateUserPassword', {
+          oldPassword: formData.oldPassword,
+          newPassword: formData.newPassword,
+        });
+        // Handle success (e.g., show a success message)
+        console.log('Password updated:', response.data);
+        setSuccessMessagePassword('Password changed successfully.'); // Set success message for password change
+        setErrors({}); // Clear any previous errors
+        setSuccessMessage(''); // Clear any previous success message for profile update
+        setFormData({ // Clear the form
+          name: '',
+          email: '',
+          oldPassword: '',
+          newPassword: '',
+        });
+      } catch (error) {
+        setErrors(error.response.data.msg); 
+        // Handle errors and update the state with validation errors
+        <Alert severity="error">{errors}</Alert>
```

Comment:
Putting an element in here won't render on the page. Only things in `return (...)` will be rendered.

Reply:
Instead, I would move this error alert to be in the card content. Maybe right under the "My Profile" header. 

```
{errors.serverMsg && <Alert severity="error">{errors.msg}</Alert>}
```
^ This will make React only render the error message _if_ `errors.serverMsg` (or whatever you choose to label errors from those catch clauses) is populated

Reply:
@akosasante akos, could you please take a look- might be doing smth wrong- I don't see custom error msg from Snackbar

Reply:
I'm not sure what you mean; the `Alert` component seems to be correctly showing the error message from the server now from what I can see:

<img width="1792" alt="Screenshot 2023-11-03 at 6 08 02 PM" src="https://github.com/Code-the-Dream-School/dd-prac-team4-front/assets/9094098/06410465-6ff7-4b85-a0e7-e0e07617a25f">

Reply:
@akosasante Akos thank you!! everything works correctly